### PR TITLE
[Java][OkHttp] Replacing Deprecated Oltu with ScribeJava

### PR DESCRIPTION
**Pull request by Valmoz**
#10661 

This PR replaces the deprecated Apache Oltu library with ScribeJava for the Java OkHttp client. 
It also tries to improve the OAuth code as follows:

- uniforming the OAuth authentication to the other auth methods, managing the Authentication as a parameter instead of using an OkHttp interceptor;
- introducing the AuthorizationCode, Password, and ClientCredentials Grants;
- managing authorization URL, refresh URL, and scope parameters for the oauth2 config.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas  @sreeshas   @jfiala   @lukoyanov   @cbornet   @jeff9finger  @karismann   @Zomzog   @lwlee2608   @nmuesch

### CODE REVIEW

**Comment by wing328** [comment]
@Valmoz thanks for the PR but looks like it broke one of our tests:
```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/others/java/okhttp-gson-streaming/src/main/java/org/openapitools/client/api/PingApi.java:[141,32] error: cannot find symbol
  symbol:   method executeStream(Call,Type)
  location: variable localVarApiClient of type ApiClient
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project petstore-okhttp-gson: Compilation failure
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/others/java/okhttp-gson-streaming/src/main/java/org/openapitools/client/api/PingApi.java:[141,32] error: cannot find symbol
[ERROR] symbol:   method executeStream(Call,Type)
[ERROR] location: variable localVarApiClient of type ApiClient
[ERROR] -> [Help 1
```

Ref: https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/13605/workflows/ec4c8e65-7dc3-4103-b278-27034353f259/jobs/27263

Can you please take a look when you've time?

**Comment by Valmoz** [comment]
@wing328  I'm sorry, I reintroduced the method I deleted by mistake.

**Comment by wing328** [comment]
Looks like there are other compliaion issues after your fix:
```
[ERROR] location: class RetryingOAuth
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/auth/RetryingOAuth.java:[159,44] error: cannot find symbol
[ERROR] symbol:   class OAuthProblemException
[ERROR] location: class RetryingOAuth
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/auth/RetryingOAuth.java:[160,22] error: reference to IOException is ambiguous
[ERROR] both constructor IOException(String) in IOException and constructor IOException(Throwable) in IOException match
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/auth/OAuthOkHttpClient.java:[31,4] error: method does not override or implement a method from a supertype
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/auth/OAuthOkHttpClient.java:[54,19] error: cannot find symbol
[ERROR] symbol:   variable OAuthClientResponseFactory
[ERROR] location: class OAuthOkHttpClient
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/okhttp-gson/src/main/java/org/openapitools/client/auth/OAuthOkHttpClient.java:[60,22] error: cannot find symbol
[ERROR] symbol:   class OAuthSystemException
[ERROR] location: class OAuthOkHttpClient
```

Ref: https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/13622/workflows/deab56df-06db-4b1e-972d-fc897a5cbc37/jobs/27280

Please take a look when you've time.

To test, run `mvn clean package` in `samples/client/petstore/java/okhttp-gson` to ensure the code compiles without issues.

File: modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pom.mustache [hunk 1]
```diff
@@ -359,7 +364,7 @@
         <jakarta.el-version>3.0.3</jakarta.el-version>
 {{/performBeanValidation}}
 {{#useBeanValidation}}
-        <beanvalidation-version>2.0.2</beanvalidation-version>
+		<beanvalidation-version>2.0.2</beanvalidation-version>
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pom.mustache:367]: Please use spaces instead of tabs.

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
Can you please also follow step 3 in the PR checklist to make sure the samples are up-to-date? Thanks.

**Comment by Valmoz** [comment]
Hi @wing328 
I should have fixed the Github account issue.

I tried to perform again step 3 (I already did in the past), and I noticed that the first command results in 4 errors (both in my branch and in _master_). I didn't modify the failing tests.

`[ERROR] Failures: 
[ERROR]   DefaultGeneratorTest.supportCustomTemplateEngine:311 » Runtime Could not gener...
[ERROR]   StringHelpersTest.startsWithSectionalTest:47->evaluate:28 » Handlebars inline@...
[ERROR]   StringHelpersTest.startsWithTest:61->evaluate:28 » Handlebars inline@43bd77be:...
[ERROR]   StringHelpersTest.startsWithYesOverrideTest:81->evaluate:28 » Handlebars inlin...
[INFO] 
[ERROR] Tests run: 1556, Failures: 4, Errors: 0, Skipped: 0`

Here you can find the error reports.

[org.openapitools.codegen.DefaultGeneratorTest.txt](https://github.com/OpenAPITools/openapi-generator/files/7467243/org.openapitools.codegen.DefaultGeneratorTest.txt)
[org.openapitools.codegen.templating.handlebars.StringHelpersTest.txt](https://github.com/OpenAPITools/openapi-generator/files/7467247/org.openapitools.codegen.templating.handlebars.StringHelpersTest.txt)

If I instead launch "mvn clean package" (without the ./mvnw file) it seems to complete without errors.
I don't know if it is due to my local settings, but I tried in my colleague's mac and he experienced the same errors.

Then I launched the generate_samples.sh and export_docs_generators.sh, but it didn't generate modifies to push to git. Could it be because the first step failed?

**Pull request closed without merging**